### ISSUE TITLE
Google Translate with Google Sheets

### ISSUE THREAD

**Issue by LiamKarlMitchell** [dannyconnell/quasalang#6]
Found a nice way to populate the translations quickly using Google Translate via Google Sheets.

Language codes can be found here.
https://cloud.google.com/translate/docs/languages

The formula entered into a new Locale column you want to add to your translations sheet.
```
=GOOGLETRANSLATE(cell with text, “source language”, “target language”)
such as
=GOOGLETRANSLATE($B2,"en","fr")
```

Then you can drag the formula down all the cells.

![image](https://user-images.githubusercontent.com/1968109/144717001-4f068671-feae-424d-b5d5-ffd132cf66a7.png)

Then you can File | Download | CSV and replace the one in your project file and generate as normal.

**Comment by fabform** [comment]
This is handy to find google sheets id
[find google sheet id](https://fabform.io/a/how-to-find-google-sheet-id)

**Comment by dannyconnell** [comment]
@LiamKarlMitchell wow thanks that's really helpful! I'm definitely gonna use this on Fudget 2.
@fabform thanks! I wonder if there's a way we can use this to automatically download the sheet as CSV (with an npm package or something) ...

**Comment by dannyconnell** [comment]
@LiamKarlMitchell we can also use an if statement to check that the source field is not blank before we do the translation, so that we can avoid errors on blank lines (if, for example we're separating different sections with blank lines and comments, as below):

<img width="515" alt="Screenshot 2022-01-10 at 14 10 23" src="https://user-images.githubusercontent.com/5763926/148780211-65012790-5d4c-48f1-926d-82fa0ba1f15f.png">

**Comment by fabform** [comment]
We could add that 
Do you know many people are using fudget 2



Sent from Yahoo Mail on Android 
 
  On Mon, 10 Jan 2022 at 14:15, Danny ***@***.***> wrote:   


@LiamKarlMitchell we can also use an if statement to check that the source field is not blank before we do the translation, so that we can avoid errors on blank lines (if, for example we're separating different sections with blank lines and comments, as below):



—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by LiamKarlMitchell** [comment]
Yeah, google sheet works great, just need to automate download from it.
If better translations are made then can just update them later, of course be careful who writes in your translations if it is going in v-html etc.

**Comment by fabform** [comment]
Handy to [find google sheet id](https://fabform.io/a/how-to-find-google-sheet-id)

**Comment by LiamKarlMitchell** [comment]
Thanks for your Quasar videos btw @dannyconnell

**Comment by LiamKarlMitchell** [comment]
Formula with check for not empty.
```=IF($B3<>"",GOOGLETRANSLATE($B3, "en", "fr"),"")```

Also you can download CSV automagically.

I was considering extending the watcher command to have an optional -gs googlesheeturl that it could poll for difference.
As an HTTP HEAD request doesn't show any kind of e-tag or version identifier sadly can only do a checksum to a temp file and over-write previous translation.csv if different as the watcher check looks at a timestamp.

If you make a google sheet public for anyone with url can view.

URL to download a google sheet looks like.
```https://docs.google.com/spreadsheets/d/{DocumentId}/gviz/tq?tqx=out:csv&sheet={SheetName}```

Or rather than extending for now, maybe I'de just make an accompanying script that does the download and re-executes quasalang seems to work well enough.
